.azure/build-pipeline.yaml

```diff
@@ -12,51 +12,111 @@ pr:
   branches:
     include:
       - '*'
-# Jobs
-jobs:
-  - job: 'build_and_test'
-    displayName: 'Build & Test'
-    # Strategy for the job
-    strategy:
-      matrix:
-        'java-11':
-          image: 'Ubuntu-18.04'
-          jdk_version: '11'
-          jdk_path: '/usr/lib/jvm/java-11-openjdk-amd64'
-    # Set timeout for jobs
-    timeoutInMinutes: 60
-    # Base system
-    pool:
-      vmImage: $(image)
-    # Variables
-    variables:
-      MVN_CACHE_FOLDER: $(HOME)/.m2/repository
-      MVN_ARGS: '-e -V -B'
-    # Pipeline steps
-    steps:
-      - task: Cache@2
-        inputs:
-          key: 'mvn-m2-cache | $(System.JobName)'
-          path: "$(MVN_CACHE_FOLDER)"
-        displayName: Maven cache
-      - template: 'templates/setup_java.yaml'
-        parameters:
-          JDK_PATH: $(jdk_path)
-          JDK_VERSION: $(jdk_version)
-      - template: 'templates/setup_minikube.yaml'
-      - bash: ".azure/scripts/build.sh"
-        env:
-          BUILD_REASON: $(Build.Reason)
-          BRANCH: $(Build.SourceBranch)
-          MVN_ARGS: "-Dquarkus.native.container-build=true"
-          DOCKER_USER: $(QUAY_USER)
-          DOCKER_PASS: $(QUAY_PASS)
-          DOCKER_REGISTRY: "quay.io"
-          DOCKER_ORG: "strimzi"
-        displayName: "Build and test"
-      - task: PublishTestResults@2
-        inputs:
-          testResultsFormat: JUnit
-          testResultsFiles: '**/TEST-*.xml'
-          testRunTitle: "Unit & Integration tests"
-        condition: always()
+
+# Stages
+stages:
+  - stage: java_build
+    displayName: Java
+    jobs:
+      - job: 'build_and_test_java'
+        displayName: 'Build & Test'
+        # Strategy for the job
+        strategy:
+          matrix:
+            'java-11':
+              image: 'Ubuntu-18.04'
```

Comment:
I think you can store this value in some variable and use it in all jobs to reduce redundancy.